src/Tempest/Database/src/Migrations/MigrationException.php

```diff
@@ -12,4 +12,14 @@ public static function noTable(): self
     {
         return new self('Migrations table does not exist. Nothing to roll back.');
     }
+
+    public static function hashMismatch(): self
```

Comment:
I prefer separate exception classes. Maybe add an interface `MigrationException` on top of all concrete migration classes.